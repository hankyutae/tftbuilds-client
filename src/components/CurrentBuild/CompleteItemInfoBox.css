.flex-box-complete-item-info{
  display:inline-flex;
  align-items:center;
  justify-content:center;
  margin:auto;
  text-align:left;

}

.item-info-box{
  display:flex;
  flex-direction:column;
  justify-content:center;
}

.item-desc{
  font-size:90%;
  max-width:300px;
  padding:3px 3px;
}