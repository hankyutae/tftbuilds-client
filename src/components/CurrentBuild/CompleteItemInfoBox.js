import React from 'react';
import './CompleteItemInfoBox.css';
import ImgLink from '../../services/create-img-link-service'

class CompleteItemInfoBox extends React.Component {
  state={
    isExpanded:this.props.isExpanded
  };
  handleExpand=()=>{
    this.setState({
      isExpanded:!this.state.isExpanded
    })
  };
  stringFixer=(string)=>{
    const arr=[];
    let stack=[];
    for(let i=0;i<string.length;i++){
      if(string[i]==='@'){
        arr.push(string[i]);
        if(stack.length===0){
          stack.push(0);
        }
        else{
          stack.pop();
        }
      }
      else{
        if(stack.length===0){
          arr.push(string[i])
        }
        else{
          arr.push(string[i].toLowerCase())
        }
      }
    }
    return arr.join('');
  }
  parseDesc=(itemInfo,item)=>{
    let namesArr=[];
    let valuesArr=[];
    itemInfo.effects.forEach(effect=>{
      namesArr.push(effect.name);
      valuesArr.push(effect.value);
    })
    let desc=this.stringFixer(itemInfo.desc);
    const problemItemDescs=[18,28,38,48,58,68,78];
    if(problemItemDescs.includes(item)){
      desc=desc.split('<br>');
      desc.shift();
      desc=desc.join('');
    }
    namesArr.forEach((name,index)=>{
      desc=desc.split(`@${name.toLowerCase()}@`).join(valuesArr[index]);
    })

    return desc;
    


  }

  render() {
    const itemInfo=this.props.items[this.props.item];
    return (
      <div className='flex-box-complete-item-info'>
        <div className='item-img-div' >
          <img className='build-item-icon' src={ImgLink.createLink(itemInfo.icon)} alt={`${itemInfo.name} icon`} onClick={(e) => { this.handleExpand(); }} />
        </div>
        {
          this.state.isExpanded && 
          <div className='item-info-box'>
            <div className='item-name'>
              {itemInfo.name}
            </div>
            <div className='item-desc'>
              {this.parseDesc(itemInfo,this.props.item)}
            </div>
          </div>
        }
      </div>
    )
  }
}

export default CompleteItemInfoBox;