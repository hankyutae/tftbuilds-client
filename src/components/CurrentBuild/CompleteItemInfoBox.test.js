import React from 'react';
import ReactDOM from 'react-dom';
import CompleteItemInfoBox from './CompleteItemInfoBox';
import items from '../TestHelpers/items';
import renderer from "react-test-renderer";


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <CompleteItemInfoBox items={items} item={11} isExpanded={true} itemIndex={0}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <CompleteItemInfoBox items={items} item={11} isExpanded={true} itemIndex={0} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
