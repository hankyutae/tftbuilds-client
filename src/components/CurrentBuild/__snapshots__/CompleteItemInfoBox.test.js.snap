// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="flex-box-complete-item-info"
>
  <div
    className="item-img-div"
    onClick={[Function]}
  >
    <img
      alt="Deathblade3 icon"
      className="build-item-icon button-cursor"
      src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_deathblade.tft.png"
    />
  </div>
  <div
    className="item-info-box"
  >
    <div
      className="item-desc"
    >
      <div>
        Whenever the wearer kills or participates in killing an enemy, gain +15 Attack Damage for the remainder of combat.
        <br />
        This effect can stack any number of times. Starts with @startingstacks@ stacks.
      </div>
    </div>
  </div>
</div>
`;
